inference-engine/src/vpu/graph_transformer/src/middleend/passes/adjust_data_batch.cpp

```diff
@@ -201,10 +206,33 @@ void PassImpl::wrapInLoop(const Model& model, const StageList& subgraph) {
         }
     }
 
+    const auto getBatchDataIndex = [](DataVector& stageInputs, const Data& data) {
+        auto i = 0;
+        for (; i < stageInputs.size(); i++) {
+            if (stageInputs[i] == data)
+                break;
+        }
+        if (i == stageInputs.size())
+            stageInputs.push_back(data);
+        return i;
+    };
+
+    // dynamic case
+    int loopStartBatchIndex = -1;
+    int loopEndBatchIndex = -1;
+    if (dynamicBatchInputs) {
+        loopStartBatchIndex = getBatchDataIndex(loopStartInputs, subgraph.front()->attrs().get<Data>("batchData"));
+        loopEndBatchIndex = getBatchDataIndex(loopEndInputs, subgraph.front()->attrs().get<Data>("batchData"));
+    }
```

Comment:
Is it possible to introduce `loopStartBatchIndex` and `loopEndBatchIndex` variables closer to a place where they are used? From diff in GitHub it seems to me it's possible..